app/v1/support/DesResponseMappingSupport.scala

```diff
@@ -33,6 +35,28 @@ trait DesResponseMappingSupport {
     }
   }
 
+  final def validateCodingOutResponse[T](desResponseWrapper: ResponseWrapper[T], taxYear: String)
+                                        (implicit currentDate: CurrentDate): Either[ErrorWrapper, ResponseWrapper[T]] = {
+    desResponseWrapper.responseData match {
+      case retrieveCodingOutDetailsResponse: RetrieveCodingOutResponse if TaxYearNotEndedValidation.validate(taxYear).isEmpty
+        && checkIfIdDoesntExists(retrieveCodingOutDetailsResponse) =>
+        Left(ErrorWrapper(desResponseWrapper.correlationId, DownstreamError))
```

Comment:
Can we add some logging to the error case here just so it's clear in Kibana where the 500 response has come from?